MovingBlocks/Terasology
# test: add NetworkMode setting to MTE

Allows MTE tests to use the `networkMode` argument to the `@IntegrationEnvironment` to specify which network mode the engine will start in: none (local player only), server with local player, or server-only (i.e. headless).

`@IntegrationEnvironment` is a new annotation. The plan is for MTE settings to move there, as opposed to making an annotation per setting. (See Terasology/ModuleTestingEnvironment#76.)

Fixes #5031

### How to Test

No impact expected on existing servers or tests, as the default is unchanged.

You could start a server (headless and headed) and connect to it just to confirm.
